src/sdk/index.js

```diff
@@ -4,23 +4,42 @@ import {merge} from 'lodash'
 import api from './api'
 import {DEFAULT_HEADERS} from './headers'
 
+function configGlobalSpec (globalSpec) {
+
+  let headers
+  if (typeof globalSpec.headers === 'function') {
+    headers = {...globalSpec.headers()}
+  }
+  else {
+    headers = globalSpec.headers
+  }
+
+  return {
+    ...globalSpec,
+    ...{headers}
+  }
+}
+
 export default function sdk (globalSpec = {}) {
 
   let defaultSpec = {
     origin: '',
     headers: DEFAULT_HEADERS
   }
-  let options = merge({}, defaultSpec, globalSpec)
+
+  let options = merge({}, defaultSpec, configGlobalSpec(globalSpec))
+
   let stream = Observable.create()
 
   function apiFactory (instanceSpec = {}) {
 
-  	return api(merge({}, options, instanceSpec, {
```

Comment:
You are correct. I was thinking of `origin` not `x-entity` and `authorization`. We would also need to add something like on line 13, so I'm doing some refactoring to try to make the pieces fit together a bit better and use a `configOptions` along with a `configGlobalSpec` because really it's the `merge` call in the `options` var that sets up the headers so I would be reluctant to name a function `getHeaders` if it is only getting the headers for globalSpec, right?